src/main/java/de/tum/in/www1/artemis/web/rest/ProgrammingExerciseExportImportResource.java

```diff
@@ -201,6 +213,37 @@ public ResponseEntity<ProgrammingExercise> importProgrammingExercise(@PathVariab
         }
     }
 
+    /**
+     * POST /programming-exercises/import-from-file: Imports an existing programming exercise from an uploaded zip file into an existing course
+     * <p>
+     * This will create the whole exercise, including all base build plans (template, solution) and repositories (template, solution, test) and copy
+     * the content from the repositories of the zip file into the newly created repositories.
+     *
+     * @param programmingExercise The exercise that should be imported
+     * @param zipFile             The zip file containing the template, solution and test repositories plus a json file with the exercise configuration
+     * @return The imported exercise (200)
+     *         (403) if the user is not at least an editor in the target course.
+     */
+    @PostMapping(IMPORT_FROM_FILE)
+    @PreAuthorize("hasRole('EDITOR')")
+    @FeatureToggle(Feature.ProgrammingExercises)
+    public ResponseEntity<ProgrammingExercise> importProgrammingExerciseFromFile(@RequestPart("programmingExercise") ProgrammingExercise programmingExercise,
+            @RequestPart("file") MultipartFile zipFile) {
+        final var user = userRepository.getUserWithGroupsAndAuthorities();
+        // Valid exercises have set either a course or an exerciseGroup
+        programmingExercise.checkCourseAndExerciseGroupExclusivity(ENTITY_NAME);
+        final var course = courseRepository.findByIdElseThrow(programmingExercise.getCourseViaExerciseGroupOrCourseMember().getId());
```

Comment:
With this I could be in the management for course `5`, but have an exercise JSON that references course `6` (I’m also instructor in). Therefore, the exercise is imported into another course.

This also expects the full course object to be present as part of the sent JSON. I don’t think this should be necessary.

I think the course ID should be a path variable from which the course is fetched and checked for access. The course in the programming exercise should be overwritten by that.